nuclear/roles/CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ foreach(rolefile ${rolefiles})
     # Roles may be within subfolders - get the subfolder and name
     file(RELATIVE_PATH role_folder ${role_source_dir} ${rolefile})
     cmake_path(REMOVE_EXTENSION role_folder)
+    string(REPLACE "/" "." role_folder ${role_folder})
```

Comment:
You're replacing a dot here but in your regex you're not accepting `.`s
Actually, now that I think about it it's not accepting `_` either...

Reply:
Yeah I was before but then I realised there were `_`s in the role names and it didn't seem to be affecting them. I guess I will see about adding them in.

Reply:
I changed them from `/` to `.` because it seemed less bad. Having them be `_` means I might not be able to recover the path

Reply:
If I'm not mistaken, the reason it accepts those with `_` when `_` isn't in the regex is that `^ROLE_[a-zA-Z0-9]+` matches anything where `[a-zA-Z0-9]` matches one character. It's not matching the whole word in that case, it's just matching until the first `_`, which is still a match. Hence, unless the first character after `ROLE_` is not included in `[a-zA-Z0-9]`, it'll keep matching without modifications.